test/functional/wallet_sendall.py

```diff
@@ -379,6 +379,63 @@ def sendall_with_maxconf(self):
         assert_equal(len(self.wallet.listunspent()), 1)
         assert_equal(self.wallet.listunspent()[0]['confirmations'], 6)
 
+    @cleanup
+    def sendall_spends_unconfirmed_change(self):
+        self.log.info("Test that sendall spends unconfirmed change")
+        self.add_utxos([17])
+        self.wallet.sendtoaddress(self.remainder_target, 10)
+        assert_greater_than(self.wallet.getbalances()["mine"]["trusted"], 6)
+        self.test_sendall_success(sendall_args = [self.remainder_target])
+
+        assert_equal(self.wallet.getbalance(), 0)
+
+    @cleanup
+    def sendall_spends_unconfirmed_inputs_if_specified(self):
+        self.log.info("Test that sendall spends specified unconfirmed inputs")
+        self.def_wallet.sendtoaddress(self.wallet.getnewaddress(), 17)
+        self.wallet.syncwithvalidationinterfacequeue()
+        assert_equal(self.wallet.getbalances()["mine"]["untrusted_pending"], 17)
+        unspent = self.wallet.listunspent(minconf=0)[0]
+
+        self.wallet.sendall(recipients=[self.remainder_target], inputs=[unspent])
+        assert_equal(self.wallet.getbalance(), 0)
+
+    @cleanup
+    def sendall_does_ancestor_aware_funding(self):
+        self.log.info("Test that sendall does ancestor aware funding for unconfirmed inputs")
+
+        # higher parent feerate
+        self.def_wallet.sendtoaddress(address=self.wallet.getnewaddress(), amount=17, fee_rate=20)
+        self.wallet.syncwithvalidationinterfacequeue()
+
+        assert_equal(self.wallet.getbalances()["mine"]["untrusted_pending"], 17)
+        unspent = self.wallet.listunspent(minconf=0)[0]
+
+        parent_txid = unspent["txid"]
+        assert_equal(self.wallet.gettransaction(parent_txid)["confirmations"], 0)
+
+        res_1 = self.wallet.sendall(recipients=[self.def_wallet.getnewaddress()], inputs=[unspent], fee_rate=20, add_to_wallet=False, lock_unspents=True)
+        child_hex = res_1["hex"]
+
+        child_tx = self.wallet.decoderawtransaction(child_hex)
+        higher_parent_feerate_amount = child_tx["vout"][0]["value"]
+
+        # lower parent feerate
+        self.def_wallet.sendtoaddress(address=self.wallet.getnewaddress(), amount=17, fee_rate=10)
+        assert_equal(self.wallet.getbalances()["mine"]["untrusted_pending"], 34)
```

Comment:
Receiving a test failure here:

```
2024-02-28T02:12:00.025000Z TestFramework (INFO): Test that sendall does ancestor aware funding for unconfirmed inputs
2024-02-28T02:12:00.174000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/path/to/bitcoin/test/functional/test_framework/test_framework.py", line 131, in main
    self.run_test()
  File "/path/to/bitcoin/test/functional/wallet_sendall.py", line 527, in run_test
    self.sendall_does_ancestor_aware_funding()
  File "/path/to/bitcoin/test/functional/wallet_sendall.py", line 20, in wrapper
    func(self)
  File "/path/to/bitcoin/test/functional/wallet_sendall.py", line 425, in sendall_does_ancestor_aware_funding
    assert_equal(self.wallet.getbalances()["mine"]["untrusted_pending"], 34)
  File "/path/to/bitcoin/test/functional/test_framework/util.py", line 57, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(17.00000000 == 34)
```

Reply:
Need to add a `syncwithvalidationinterfacequeue()` after the send call. The mempool event that notifies the second wallet is consumed by a different thread (not the http one).

Reply:
Should be fixed now, sorry for the delay.